test/libsolidity/SemanticTest.cpp

```diff
@@ -452,3 +457,16 @@ std::optional<bytes> SemanticTest::builtinSmokeTest(FunctionCall const& call)
 	}
 	return result;
 }
+
+std::optional<bytes> SemanticTest::builtinSmokeReactionTest(FunctionCall const& call)
```

Comment:
Why is this needed/

Reply:
I just added this to be able to test the "reactions". The name of this builtin is `smoke.reaction`, where the `SmokeHook` is setting a test-reaction for exactly that builtin.

Reply:
I think it is fine to test them with actual events once we have that.